Editor/Overlays/CinemachineToolbarOverlay.cs

```diff
@@ -1,15 +1,213 @@
 using UnityEngine;
 using UnityEngine.UIElements;
+using System.Collections.Generic;
 
 namespace Cinemachine.Editor
 {
+#if UNITY_2022_1_OR_NEWER
     /// <summary>
-    /// To display a CinemachineExclusiveEditorToolbarToggle in the Cinemachine Toolbar,
-    /// TODO: Make this extendable when another PR lands (see: CMCL-501),
+    /// This is a generic Tool class for Cinemachine tools.
+    /// To create a new tool, inherit from CinemachineTool and implement GetIcon().
+    /// Your new tool will need to have the [EditorTool("tool-name", typeof(CinemachineVirtualCameraBase))] attribute.
+    ///
+    /// A tool will be drawn iff it has been registered using CinemachineSceneToolUtility.RegisterTool.
+    /// This is generally done in the OnEnable function of the editor script of the cinemachine component
+    /// (CinemachineVirtualCamera, CinemachineComponentBase), for which the tool was meant.
+    /// To unregister, call CinemachineSceneToolUtility.UnregisterTool in the same script's OnDisable function.
+    ///
+    /// To draw the handles related to the tool, you need to implement your drawing function and call it in the
+    /// editor script's OnSceneGUI function. An alternative for drawing handles is to override this function's
+    /// OnToolGUI or OnDrawHandles functions (see EditorTool or IDrawSelectedHandles docs for more information).
+    ///
+    /// To check, if a tool has been enabled/disabled in the editor script, use CinemachineSceneToolUtility.IsToolActive.
+    /// </summary>
+    public abstract class CinemachineTool : EditorTool, IDrawSelectedHandles
+    {
+        GUIContent m_IconContent;
+
+        /// <summary>
+        /// Implement this to set your Tool's icon and tooltip.
+        /// </summary>
+        /// <returns>A GUIContent with an icon set.</returns>
+        protected abstract GUIContent GetIcon();
+
+        /// <summary>This lets the editor find the icon of the tool.</summary>
+        public override GUIContent toolbarIcon => m_IconContent ??= GetIcon();
+
+        // <summary>This is called when the Tool is selected in the editor.</summary>
+        public override void OnActivated()
+        {
+            base.OnActivated();
+            CinemachineSceneToolUtility.SetTool(true, GetType());
+        }
+
+        // <summary>This is called when the Tool is deselected in the editor.</summary>
+        public override void OnWillBeDeactivated()
+        {
+            base.OnWillBeDeactivated();
+            CinemachineSceneToolUtility.SetTool(false, GetType());
+        }
+        
+        // <summary>This checks whether this tool should be displayed or not.</summary>
+        public override bool IsAvailable()
+        {
+            return CinemachineSceneToolUtility.IsToolRequired(GetType());
+        }
+        
+        // Implement IDrawSelectedHandles to draw gizmos for this tool even if it is not the active tool.
+        public void OnDrawHandles()
+        {
+        }
```

Comment:
XML comments require /// not //.  
Also params and return values need to be documented, or there will be doc errors in ci.
Also XML tag missing on the last method.